### DISCUSSION TITLE
CLOG-GUI Destop Progress

### DISCUSSION THREAD

**Discussion by rabbibotton** [rabbibotton/clog#27] [Show and tell]
The clog-gui desktop abstraction has come along very nicely, I am spending more time on it then I expected to as I see the great advantage to having it ready for upcoming tools. It is very usable as is, I am just adding here there some additional windowing menus and dialog boxes for load and save (for local file :)

Long term I will circle back for stronger iPhone/Android support but that is not really the target for this extension of CLOG.

**Comment by rabbibotton** [comment]
<img width="928" alt="clip" src="https://user-images.githubusercontent.com/11616789/108012356-28514a80-6fd7-11eb-8590-aaf796023ede.png">

I have added to demo3 functional save as and open dialog boxes. I will be added them soon to clog-gui

**Comment by ghost** [comment]
It reminds me a little of Corman Lisp's UI, but sleeker.

**Reply by rabbibotton** [reply]
Cool. I haven't seen their UI but I guess that makes me a great mind to think a like :)

**Comment by rabbibotton** [comment]
The dialog moved to clog-gui and small little enhancements to it.

**Comment by rabbibotton** [comment]
<img width="822" alt="Screen Shot 2021-02-16 at 10 41 03 PM 1" src="https://user-images.githubusercontent.com/11616789/108152982-3de27500-70a8-11eb-87d6-cb149c3ea37f.png">

So now supports modal dialog boxes and stay-on-top windows.

**Comment by rabbibotton** [comment]
Added some more dialog boxes about-dialog and input-dialog. Also move to a new package clog-gui as this is an extension of clog.

**Comment by rabbibotton** [comment]
Added a new standard dialog - a form-dialog - you give it an a-list of field descriptions to fieldnames and it returns an a-list of fieldnames to fieldvalues have an example of all the dialogs in tutorial 22.

**Comment by rabbibotton** [comment]
Added to form-dialogs ability to handle select lists, radio buttons, check boxes etc
Added "Toasts" with and without timeouts. See tutorial 22 for examples.
